From bunk@stusta.de Sat Oct  7 21:32:54 2006
Date: Sun, 8 Oct 2006 06:06:38 +0200
From: Adrian Bunk <bunk@stusta.de>
To: ebiederm@xmission.com
Cc: linux-kernel@vger.kernel.org, Jesper Juhl <jesper.juhl@gmail.com>, gregkh@suse.de, linux-pci@atrey.karlin.mff.cuni.cz
Subject: PCI: HT_IRQ must depend on PCI
Message-ID: <20061008040638.GF29474@stusta.de>
Content-Disposition: inline

CONFIG_PCI=n, CONFIG_HT_IRQ=y results in the following compile error:

<--  snip  -->

...
  LD      vmlinux
arch/i386/mach-generic/built-in.o: In function `apicid_to_node':
summit.c:(.text+0x53): undefined reference to `apicid_2_node'
arch/i386/kernel/built-in.o: In function `arch_setup_ht_irq':
(.text+0xcf79): undefined reference to `write_ht_irq_low'
arch/i386/kernel/built-in.o: In function `arch_setup_ht_irq':
(.text+0xcf85): undefined reference to `write_ht_irq_high'
arch/i386/kernel/built-in.o: In function `k7nops':
alternative.c:(.data+0x1358): undefined reference to `mask_ht_irq'
alternative.c:(.data+0x1360): undefined reference to `unmask_ht_irq'
make[1]: *** [vmlinux] Error 1

<--  snip  -->

Bug report by Jesper Juhl.

Signed-off-by: Adrian Bunk <bunk@stusta.de>
Signed-off-by: Eric W. Biederman <ebiederm@xmission.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/Kconfig |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/pci/Kconfig
+++ gregkh-2.6/drivers/pci/Kconfig
@@ -55,7 +55,7 @@ config PCI_DEBUG
 config HT_IRQ
 	bool "Interrupts on hypertransport devices"
 	default y
-	depends on X86_LOCAL_APIC && X86_IO_APIC
+	depends on PCI && X86_LOCAL_APIC && X86_IO_APIC
 	help
 	   This allows native hypertransport devices to use interrupts.
 
